util/src/test/java/io/zeebe/util/SocketUtilTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.zeebe.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import org.junit.Test;
+
+public class SocketUtilTest {
```

Comment:
```suggestion
public final class SocketUtilTest {
```

Reply:
Can we also add a comment about how it relies on `-Djdk.net.hosts.file=src/test/resources/test-hosts` as specified in the pom? Meaning it probably won't work via the IDE I guess.

Actually, is there a way we could make this work so it would also work in the IDE? Would mocking it be the only way? I'm not asking for it, as for such a utility class I think it'd make the test too artificial...but maybe you're aware of something I'm not :slightly_smiling_face:

Reply:
Hey, I was a little bit thinking about this and consider the next options:
1. Use reflection to actually set proper implementation of the `java.net.InetAddress.NameService`. I think this is a bit ugly, I would like not to do this 😅
2. We can create a shared IDEA configuration, that actually sets a correct option. But it hard to maintain: you can't put this on CI, if you will rename/move the file you can't run this configuration any longer until you will fix it.